src/backend/src/services/design-reviews.services.ts

```diff
@@ -0,0 +1,126 @@
+   * @param designReviewId the id of the design review to edit
+   * @param dateScheduled the date of the design review
+   * @param teamTypeId the team that the design_review is for (software, electrical, etc.)
+   * @param requiredMembers required members for the design review
+   * @param optionalMembers optional members for the design review
+   * @param isOnline is the design review online (IF TRUE: zoom link should be requried + inPerson must be false)
+   * @param isInPerson is the design review in person (IF TRUE: location should be required)
+   * @param zoomLink the zoom link for the design review meeting
+   * @param location the location for the design review meeting
+   * @param docTemplateLink the document template link for the design review
+   * @param status see Design_Review_Status enum
+   * @param confirmedMembers confirmed members for the design review (cannot be in deniedMembers)
+   * @param deniedMembers denied members for the design review (cannot be in confirmedMembers)
+   * @param attendees the attendees for the design review (should they have any relation to the other shit / can't edit this after STATUS: DONE)
+   * @param meetingTimes meeting time must be between 0-48 (9-9, 15 minute increments)
+   */
+
+  static async editDesignReview(
+    user: User,
+    designReviewId: string,
+    dateScheduled: Date,
+    teamTypeId: string,
+    requiredMembers: number[],
+    optionalMembers: number[],
+    isOnline: boolean,
+    isInPerson: boolean,
+    zoomLink: string | null,
+    location: string | null,
+    docTemplateLink: string | null,
+    status: Design_Review_Status,
+    confirmedMembers: number[],
+    deniedMembers: number[],
+    attendees: number[],
+    meetingTimes: number[]
+  ): Promise<Design_Review> {
+    // verify user is allowed to edit work package
+    if (isNotLeadership(user.role)) throw new AccessDeniedMemberException('edit design reviews');
+
+    // make sure the confirmedMembers are not in the denied Members
+    // for review: should someone be able to edit confifirmed and denied members?
+    if (confirmedMembers.length > 0 && confirmedMembers.some((cMember) => deniedMembers.includes(cMember))) {
+      throw new HttpException(400, 'confirmed members cannot be in denied members');
+    }
+
+    // make sure the requiredMembers are not in the optionalMembers
+    if (requiredMembers.length > 0 && requiredMembers.some((rMember) => optionalMembers.includes(rMember))) {
+      throw new HttpException(400, 'required members cannot be in optional members');
+    }
+
+    // make sure the design review is either online or (exclusive) in person
+    if (isOnline && isInPerson) {
+      throw new HttpException(400, 'design review cannot be both online and in person');
+    }
+    // make sure there is a zoom link if the design review is online
+    // (zoom link is presumed to be greater then 2 char) is there a better way to test this?
```

Comment:
using .isUrl() in the route should be enough, esp since we're storing these as strings not as Link objects

Reply:
The comment specifying that the zoom link is greater then 2 chars was actually not enforced.  
That being said, is it necessary to enforce the user adding a zoomLink / Location at all?